lib/Gedmo/Tree/Entity/Repository/NestedTreeRepository.php

```diff
@@ -193,6 +193,114 @@ public function getPath($node)
         return $this->getPathQuery($node)->getResult();
     }
 
+    /**
+     * @see getAncestorsQueryBuilder
+     */
+    public function ancestorsQueryBuilder($node = null, $direct = false, $sortByField = null, $direction = 'ASC', $includeNode = false)
+    {
+        $meta = $this->getClassMetadata();
+        $config = $this->listener->getConfiguration($this->_em, $meta->name);
+
+        $qb = $this->getQueryBuilder();
+        $qb->select('node')
+            ->from($config['useObjectClass'], 'node')
+        ;
+        if ($node !== null) {
+            if ($node instanceof $meta->name) {
+                $wrapped = new EntityWrapper($node, $this->_em);
+                if (!$wrapped->hasValidIdentifier()) {
+                    throw new InvalidArgumentException("Node is not managed by UnitOfWork");
+                }
+                if ($direct) {
+                    $qb->where($qb->expr()->eq('node.'.$config['parent'], ':pid'));
+                    $qb->setParameter('pid', $wrapped->getIdentifier());
+                } else {
+                    $left = $wrapped->getPropertyValue($config['left']);
+                    $right = $wrapped->getPropertyValue($config['right']);
+                    if ($left && $right) {
+                        $qb->where($qb->expr()->gt('node.'.$config['right'], $right));
+                        $qb->andWhere($qb->expr()->lt('node.'.$config['left'], $left));
+                    }
+                }
+                if (isset($config['root'])) {
+                    $qb->andWhere($qb->expr()->eq('node.'.$config['root'], ':rid'));
+                    $qb->setParameter('rid', $wrapped->getPropertyValue($config['root']));
+                }
+                if ($includeNode) {
+                    $idField = $meta->getSingleIdentifierFieldName();
+                    $qb->where('('.$qb->getDqlPart('where').') OR node.'.$idField.' = :rootNode');
+                    $qb->setParameter('rootNode', $node);
+                }
+            } else {
+                throw new \InvalidArgumentException("Node is not related to this repository");
+            }
+        } else {
+            if ($direct) {
+                $qb->where($qb->expr()->isNull('node.'.$config['parent']));
+            }
+        }
+        if (!$sortByField) {
+            $qb->orderBy('node.'.$config['left'], 'DESC');
+        } elseif (is_array($sortByField)) {
+            $fields = '';
```

Comment:
Since `$sortByField` could never be an empty array given the previous condition, it could be simplified as `$fields = 'node.'.implode(', node.', $sortByField);`.